apps/cowswap-frontend/src/api/gnosisProtocol/api.ts

```diff
@@ -141,6 +141,16 @@ function _mapNewToLegacyParams(params: FeeQuoteParams): OrderQuoteRequest {
 export async function getQuote(params: FeeQuoteParams): Promise<OrderQuoteResponse> {
   const { chainId } = params
   const quoteParams = _mapNewToLegacyParams(params)
+  const { sellToken, buyToken } = quoteParams
+
+  if (sellToken === buyToken) {
+    return Promise.reject(
+      mapOperatorErrorToQuoteError({
+        errorType: ApiErrorCodes.SameBuyAndSellToken,
+        description: GpQuoteErrorDetails.SameBuyAndSellToken,
+      })
+    )
+  }
```

Comment:
Right, but this is the API.
I saw that we were doing one request sometimes with he same sell/buy
This is just preventing to post it to the API

Also you found a bug that reproduces it (by changing networks), so the query will still fail, but we don't punish the API. Also because in some places we might have some bad error handling, we re-attempt too often. 

So this change was just a nit i added